qa/rpc-tests/txoutsbyaddress.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+# Copyright (c) 2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+
+class TxOutsByAddressTest(BitcoinTestFramework):
+    """Tests -txoutsbyaddressindex"""
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 3
+
+    def setup_network(self, split=False):
+        print("Setup network...")
+        self.nodes = []
+        self.nodes.append(start_node(0, self.options.tmpdir, ["-txoutsbyaddressindex"]))
+        self.nodes.append(start_node(1, self.options.tmpdir))
+        self.nodes.append(start_node(2, self.options.tmpdir))
+        self.is_network_split = False
+        self.sync_all()
+
+    def run_test(self):
+        print("Generating test blockchain...")
+
+        # Check that there's no UTXO on any of the nodes
+        for node in self.nodes:
+            assert_equal(len(node.listunspent()), 0)
+
+        # mining
+        connect_nodes(self.nodes[1], 0)
+        connect_nodes(self.nodes[2], 0)
+        self.nodes[0].generate(101)
+        self.sync_all()
+        assert_equal(self.nodes[0].getbalance(), 50)
+
+        # TX1: send from node0 to node1
+        # - check if txout from tx1 is there
+        address = self.nodes[1].getnewaddress()
+        txid1 = self.nodes[0].sendtoaddress(address, 10)
+        self.nodes[0].generate(101) # node will collect its own fee
+        self.sync_all()
+        assert_equal(self.nodes[0].getbalance(), 5090)
+        assert_equal(self.nodes[1].getbalance(), 10)
+        txouts = self.nodes[0].gettxoutsbyaddress(1, (address,))
+        txid = txouts[0]["txid"]
+        assert_is_hash_string(txid)
+        assert_equal(txid, txid1)
+
+        # stop node 2
```

Comment:
node 2 is stopped so that later on it can be restarted and then orphan a block created by node 1 (to test txoutindex handles the reorg)

basically its stopped so I will not build on a block created by node 1